include/os/linux/kernel/linux/simd_x86.h

```diff
@@ -135,7 +135,10 @@
  */
 #if defined(HAVE_KERNEL_FPU_INTERNAL)
 
+#include <sys/zfs_context.h>
+
 extern union fpregs_state **zfs_kfpu_fpregs;
+extern kmem_cache_t *zfs_kfpu_state_cache;
```

Comment:
We should use `struct kmem_cache` here, the `kmem_cache_t` type is provided by the spl for the illumos slab implementation.  

Unfortunately, the illumos and kernel slab APIs conflict with each other so there's some `#define` tricky in `sys/kmem_cache.h` to map conflicting symbols to the spl implementation.  

```c
#define kmem_cache_create(name, size, align, ctor, dtor, rclm, priv, vmp, fl) \
    spl_kmem_cache_create(name, size, align, ctor, dtor, rclm, priv, vmp, fl)
#define kmem_cache_destroy(skc)         spl_kmem_cache_destroy(skc)
#define kmem_cache_alloc(skc, flags)    spl_kmem_cache_alloc(skc, flags)
#define kmem_cache_free(skc, obj)       spl_kmem_cache_free(skc, obj)
```

Reply:
see above